packages/beacon-node/src/chain/initState.ts

```diff
@@ -118,7 +118,7 @@ export async function initStateFromEth1({
 
     return genesisResult.state;
   } catch (e) {
-    if (builder.lastProcessedBlockNumber != null) {
+    if (builder.lastProcessedBlockNumber !== null) {
```

Comment:
Since this is has a slightly different meaning, I'd go through each one carefully to make sure no bug is introduced.
Double check each case in this PR that looked like `foo != null` or `foo == null` to make sure that the proper behavior is preserved. If `foo` can be either `null` or `undefined` then we need to check each case (or keep `== null`). If `foo` is either `null` OR `undefined` then it should be good to change and tighten the condition.

Reply:
I agree each diff should be carefully inspected. Also will it be possible to run this branch on our tests servers for a while before merging?